pkg/cloud/gcp/actuators/machineset/controller.go

```diff
@@ -175,6 +176,15 @@ func (r *Reconciler) reconcile(machineSet *machinev1.MachineSet) (ctrl.Result, e
 	machineSet.Annotations[labelsKey] = mapiutil.MergeCommaSeparatedKeyValuePairs(
 		fmt.Sprintf("kubernetes.io/arch=%s", util.CPUArchitecture(providerConfig.MachineType)),
 		machineSet.Annotations[labelsKey])
+
+	// When upgrading from 4.12 on GCP marketplace, the MachineSets refer to images that do not support shielded VMs.
+	// However, GCP defaults to shielded VMs sometime between 4.12 and 4.13.
+	// Therefore, we should disable the shielded instance config in the MachineSet's template, so that new Machines created from it will boot.
+	if has48MarketplaceImage(providerConfig.Disks) && providerConfig.ShieldedInstanceConfig == (machinev1.GCPShieldedInstanceConfig{}) {
+		providerConfig.ShieldedInstanceConfig.SecureBoot = machinev1.SecureBootPolicyDisabled
+		providerConfig.ShieldedInstanceConfig.VirtualizedTrustedPlatformModule = machinev1.VirtualizedTrustedPlatformModulePolicyDisabled
+		providerConfig.ShieldedInstanceConfig.IntegrityMonitoring = machinev1.IntegrityMonitoringPolicyDisabled
+	}
```

Comment:
I expect these settings to be consumed [here](https://github.com/openshift/machine-api-provider-gcp/blob/main/pkg/cloud/gcp/actuators/machine/reconciler.go#L199-L209) on machine creation.